atlasdb-container-test-utils/src/main/java/com/palantir/atlasdb/containers/DockerProxyCqlSessionBuilder.java

```diff
@@ -0,0 +1,137 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.containers;
+
+import java.net.SocketAddress;
+
+import javax.annotation.Nonnull;
+
+import com.datastax.oss.driver.api.core.CqlSessionBuilder;
+import com.datastax.oss.driver.api.core.config.DriverConfigLoader;
+import com.datastax.oss.driver.api.core.context.DriverContext;
+import com.datastax.oss.driver.api.core.session.ProgrammaticArguments;
+import com.datastax.oss.driver.internal.core.context.DefaultDriverContext;
+import com.datastax.oss.driver.internal.core.context.NettyOptions;
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.buffer.ByteBufAllocator;
+import io.netty.channel.Channel;
+import io.netty.channel.EventLoopGroup;
+import io.netty.handler.proxy.Socks5ProxyHandler;
+import io.netty.util.Timer;
+import io.netty.util.concurrent.EventExecutorGroup;
+import io.netty.util.concurrent.Future;
+
+/**
+ * Implementation of a {@link CqlSessionBuilder} used for testing. This builder sets up the
+ * {@link com.datastax.oss.driver.internal.core.context.InternalDriverContext} such that {@link NettyOptions} forward
+ * all communication to the specified proxy passed during builder creation.
+ */
+final class DockerProxyCqlSessionBuilder extends CqlSessionBuilder {
+    private final SocketAddress proxyAddress;
+
+    DockerProxyCqlSessionBuilder(SocketAddress proxyAddress) {
+        this.proxyAddress = proxyAddress;
+    }
+
+    @Override
+    protected DriverContext buildContext(
+            DriverConfigLoader configLoader,
+            ProgrammaticArguments programmaticArguments) {
+        return new DockerProxyDriverContext(configLoader, programmaticArguments, proxyAddress);
+    }
+
+    /**
+     * This class follows the intended usage explained by {@link DefaultDriverContext}.
```

Comment:
nit: `explained in the docs of DefaultDriverContext`

Reply:
Added.